de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java

```diff
@@ -1,507 +1,546 @@
+		return table;
+	}
+	
+	/**
+	 * @param item the editor page
+	 * @param tableName the name of the table
+	 * @return it returns the desired SWTBotTable
+	 */
+	protected SWTBotTable getSWTBotTable(SWTBotTreeItem item, String tableName) {
+		openEditor(item);
+		SWTBotSection composite = getSWTBotSection(tableName);
+		SWTBotTable table = composite.getSWTBotTable();
+		return table;
+	}
+	
+	/**
+	 * @param clazz the class name of the section
+	 * @return it returns the desired section name
+	 */
+	protected String getSectionName(Class<?> clazz) {
+		return "Section for: " + clazz.getSimpleName();
+	}
+	
+	/**
+	 * @param clazz the class name of the section
+	 * @return it returns the desired section
+	 */
+	protected SWTBotSection getSWTBotSection(Class<?> clazz) {
+		String sectionName = getSectionName(clazz);
+		return getSWTBotSection(sectionName);
+	}
+	
+	/**
+	 * @param sectionName the name of the section
+	 * @return it returns the desired section
+	 */
+	protected SWTBotSection getSWTBotSection(String sectionName) {
+		@SuppressWarnings("unchecked")
+		Matcher<Section> matcher = allOf(widgetOfType(Section.class), withMnemonic(sectionName));
+		SWTBotSection composite = new SWTBotSection(bot.widget(matcher, 0), matcher);
+		return composite;
+	}
+	
+	
+	/**
+	 * @param table the table to make changes on
+	 * @param rowPosition the row positon on the table
+	 * @param columnPosition the column position on the table
+	 * @param oldValue the old value to change
+	 * @param newValue the new value
+	 */
+	protected void setTableValue(SWTBotTable table, int rowPosition, int columnPosition, String oldValue, String newValue) {
+		table.click(rowPosition, columnPosition);
+		bot.text(oldValue).setText(newValue);		
+	}
+	/**
+	 * Opens the inheritance dialog and selects the element 
+	 * @param element the element that inherits
+	 * @param rootName the name of the root object which contains the inheriting element
+	 * @param elementNames the path towards to inheriting object from root object
+	 */
+	public void addInheritance(SWTBotTreeItem element, String rootName, String... elementNames) {
+		openEditor(element);
+		bot.button("Add Inheritance").click();
+		
+		SWTBotTreeItem dialogueElement = bot.tree().getTreeItem(rootName);
+		
+		for (String elementName : elementNames) {
+			dialogueElement.expand();
+			dialogueElement = dialogueElement.getNode(elementName);
+		}
+		
+		dialogueElement.select();
+		
+		bot.button("OK").click();
+		waitForEditingDomainAndUiThread();
+	}
+	
+	/**
+	 * returns the repository
+	 * @param project the project
+	 * @return rep the repository
+	 * @author bell_er
+	 *
+	 */
+	public static Repository getRepository(IProject project) {
+		VirSatResourceSet resSetRepositoryTarget = VirSatResourceSet.getResourceSet(project);
+		Repository rep = resSetRepositoryTarget.getRepository();
+		return rep;
+		
+	}
+	
+	/**
+	 * A Runnable lock to make sure that the display thread executed all messages
+	 * @author fisc_ph

```

Comment:
Didn't we want to stop adding author tags?

Reply:
true, but since there are so many already, we should have a ticket and wipe them away. I create a ticket for that #268.